src/NuGet.CommandLine/Commands/PushCommand.cs

```diff
@@ -126,10 +128,10 @@ private void PushPackage(string packagePath, string source, string apiKey, TimeS
 
             EnsurePackageFileExists(packagePath, packagesToPush);
 
-            foreach (string packageToPush in packagesToPush)
+            Parallel.ForEach(packagesToPush, packageToPush =>
```

Comment:
We should control the max amount of parallel pushes, and probably have a top level parameter to control it.